Source/SvgElementIdManager.cs

```diff
@@ -153,15 +153,6 @@ public string EnsureValidId(string id, bool autoForceUniqueID = false)
                 return id;
             }
 
-            if (char.IsDigit(id[0]))
-            {
-                if (autoForceUniqueID)
-                {
-                    return EnsureValidId("id" + id, true);
-                }
-                throw new SvgIDWrongFormatException("ID cannot start with a digit: '" + id + "'.");
-            }
-
             if (this._idValueMap.ContainsKey(id))
             {
                 if (autoForceUniqueID)

```

Comment:
Here, if an ID is duplicated, a unique ID is generated.
`41cb5de8-82e7-4ab5-94d7-87e10775c848` -> `41cb5de8-82e7-4ab5-94d7-87e10775c848#1`, `41cb5de8-82e7-4ab5-94d7-87e10775c848#2` ...

Reply:
However, this can not keep perfect uniqueness.

```svg
<svg version="1.1" baseProfile="basic" id="svg-root"
  width="100%" height="100%" viewBox="0 0 480 360"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <g id="test-body-content" font-family="SVGFreeSansASCII,sans-serif" font-size="18">
    <linearGradient id="Grad1a" gradientUnits="objectBoundingBox" x1="0" y1="0" x2="1" y2="0">
      <stop stop-color="blue" offset="0"/>
      <stop stop-color="lime" offset="1"/>
    </linearGradient>
    <!-- this is referenced... -->
    <linearGradient id="Grad1a" gradientUnits="objectBoundingBox" x1="0" y1="0" x2="1" y2="0">
      <stop stop-color="green" offset="0"/>
      <stop stop-color="yellow" offset="1"/>
    </linearGradient>
    <!-- correctly this is referenced. -->
    <linearGradient id="Grad1a#1" gradientUnits="objectBoundingBox" x1="0" y1="0" x2="1" y2="0">
      <stop stop-color="red" offset="0"/>
      <stop stop-color="white" offset="1"/>
    </linearGradient>
    <rect x="20" y="20" width="440" height="80" fill="url(#Grad1a#1)"/>
    <text font-size="30" x="20" y="130">Linear gradient.</text>
  </g>
</svg>
```